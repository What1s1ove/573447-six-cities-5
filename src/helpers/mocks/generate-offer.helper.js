import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
