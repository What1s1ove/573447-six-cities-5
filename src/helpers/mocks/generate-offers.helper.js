import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
