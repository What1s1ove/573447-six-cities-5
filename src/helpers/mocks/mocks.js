export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
