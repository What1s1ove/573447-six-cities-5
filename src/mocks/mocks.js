export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
