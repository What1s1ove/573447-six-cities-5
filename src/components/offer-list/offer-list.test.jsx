import * as React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'enzyme';
import {mockedOffers} from '~/mocks/mocks';
import OfferList from './offer-list';

describe(`OfferList component`, () => {
  it(`should render OfferList component`, () => {
    const component = render(
        <MemoryRouter>
          <OfferList
            className=""
            offers={mockedOffers}
            onActiveOfferChange={jest.fn()}
            onFavoriteToggle={jest.fn()}
          />
        </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
