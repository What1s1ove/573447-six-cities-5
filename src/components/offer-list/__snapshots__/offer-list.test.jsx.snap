// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList component should render OfferList component 1`] = `
<div
  class="places__list"
>
  <article
    class="cities__place-card place-card"
  >
    <div
      class="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      class="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
      >
        <img
          alt="Place image"
          class="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      class="place-card__info"
    >
      <div
        class="place-card__price-wrapper"
      >
        <div
          class="place-card__price"
        >
          <b
            class="place-card__price-value"
          >
            €241
          </b>
          <span
            class="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          class="place-card__bookmark-button button"
          type="button"
        >
          <svg
            class="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        class="place-card__rating rating"
      >
        <div
          class="place-card__stars rating__stars"
        >
          <span
            style="width:80%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        class="place-card__name"
      >
        <a
          href="/offer/1"
        >
          The Joshua Tree House
        </a>
      </h2>
      <p
        class="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    class="cities__place-card place-card"
  >
    <div
      class="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
      >
        <img
          alt="Place image"
          class="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      class="place-card__info"
    >
      <div
        class="place-card__price-wrapper"
      >
        <div
          class="place-card__price"
        >
          <b
            class="place-card__price-value"
          >
            €909
          </b>
          <span
            class="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          class="place-card__bookmark-button button"
          type="button"
        >
          <svg
            class="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        class="place-card__rating rating"
      >
        <div
          class="place-card__stars rating__stars"
        >
          <span
            style="width:100%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        class="place-card__name"
      >
        <a
          href="/offer/2"
        >
          Loft Studio in the Central Area
        </a>
      </h2>
      <p
        class="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    class="cities__place-card place-card"
  >
    <div
      class="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
      >
        <img
          alt="Place image"
          class="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      class="place-card__info"
    >
      <div
        class="place-card__price-wrapper"
      >
        <div
          class="place-card__price"
        >
          <b
            class="place-card__price-value"
          >
            €191
          </b>
          <span
            class="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          class="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            class="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        class="place-card__rating rating"
      >
        <div
          class="place-card__stars rating__stars"
        >
          <span
            style="width:80%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        class="place-card__name"
      >
        <a
          href="/offer/3"
        >
          The Pondhouse - A Magical Place
        </a>
      </h2>
      <p
        class="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;
